packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -480,56 +805,77 @@ const DialogueCheckBox: React.FC<{
   );
 };
 
-const isMatched = (userDialogueChecks: DialogueCheckInfo[], targetUserId: string): boolean => {
-  return userDialogueChecks.some(check => check.targetUserId === targetUserId && check.match);
-};
-
-const isChecked = (userDialogueChecks: DialogueCheckInfo[], targetUserId: string): boolean => {
-  return userDialogueChecks?.find(check => check.targetUserId === targetUserId)?.checked || false;
-};
-
-type MatchDialogueButtonProps = {
-  isMatched: boolean;
-  targetUserId: string;
-  targetUserDisplayName: string;
-  currentUser: UsersCurrent;
-  loadingNewDialogue: boolean;
-  createDialogue: (title: string, participants: string[]) => void;
-  classes: ClassesType;
-};
-
 const MatchDialogueButton: React.FC<MatchDialogueButtonProps> = ({
   isMatched,
+  checkId,
   targetUserId,
   targetUserDisplayName,
   currentUser,
-  loadingNewDialogue,
-  createDialogue,
   classes,
 }) => {
+
+  const { openDialog } = useDialog();
+  const { history } = useNavigation();
+
+  const {loading: userLoading, results} = useMulti({
+    terms: {
+      view: "dialogueMatchPreferences",
+      dialogueCheckId: checkId,
+      limit: 1000,
+    },
+    fragmentName: "DialogueMatchPreferenceInfo",
+    collectionName: "DialogueMatchPreferences",
+  });
+
   if (!isMatched) return <div></div>; // need this instead of null to keep the table columns aligned
 
-  return (
-    <div>
+  const userMatchPreferences = results && results.length > 0 ? results[0] : null;
+  const generatedDialogueId = userMatchPreferences?.generatedDialogueId;
+
+  const renderButton = () => {
+    if (!!generatedDialogueId) {
+      return (
+        <button className={classes.lightGreenButton} onClick={(e) => redirect(generatedDialogueId, history)}>
+          <a data-cy="message">Go to dialogue</a>
+        </button>
+      );
+    }
+  
+    if (userMatchPreferences) {
+      return (
+        <button className={classes.waitingButton} disabled>
```

Comment:
When writing new code, please avoid using raw <button> elements and instead use Material UI buttons as we do most places

Reply:
Any particular reason? I think raw elements are great! Most importantly, they're much easier for shoggoth to work with, which speeds up our iteration loop a lot.

Reply:
Styling & behavior remains consistent with everything else in the codebase.  I don't actually know what specific differences to expect in this case, I just know that in practice there often are differences and sometimes you'll spend a whole bunch of time trying to debug some issue that turned out to be downstream of an inconsistency like this